internal/dirpool/resource_dirpool.go

```diff
@@ -0,0 +1,224 @@
+	rrSetKey := rrset.GetRRSetKeyFromID(rd.Id())
+	rrSetKey.PType = sdkpool.DIR
+	_, resList, err := services.RecordService.Read(rrSetKey)
+
+	if err != nil {
+		rd.SetId("")
+
+		return nil
+	}
+
+	if len(resList.RRSets) > 0 {
+		if err = flattenDIRPool(resList, rd); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return diags
+}
+
+func resourceDIRPoolUpdate(ctx context.Context, rd *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	services := meta.(*service.Service)
+	rrSetData := getNewDIRPoolRRSet(rd)
+	rrSetKeyData := rrset.GetRRSetKeyFromID(rd.Id())
+
+	_, err := services.RecordService.Update(rrSetKeyData, rrSetData)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceDIRPoolRead(ctx, rd, meta)
+}
+
+func resourceDIRPoolDelete(ctx context.Context, rd *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	services := meta.(*service.Service)
+	rrSetKeyData := rrset.GetRRSetKeyFromID(rd.Id())
+
+	_, err := services.RecordService.Delete(rrSetKeyData)
+
+	if err != nil {
+		rd.SetId("")
+
+		return diag.FromErr(err)
+	}
+
+	rd.SetId("")
+
+	return diags
+}
+
+func getNewDIRPoolRRSet(rd *schema.ResourceData) *sdkrrset.RRSet {
+	rrSetData := rrset.NewRRSetWithRecordDataInfo(rd)
+	profile := &dirpool.Profile{}
+	rrSetData.Profile = profile
+
+	if val, ok := rd.GetOk("pool_description"); ok {
+		profile.Description = val.(string)
+	}
+
+	if val, ok := rd.GetOk("conflict_resolve"); ok {
+		profile.ConflictResolve = val.(string)
+	}
+
+	if val, ok := rd.GetOk("ignore_ecs"); ok {
+		profile.IgnoreECS = val.(bool)
+	}
+
+	if val, ok := rd.GetOk("no_response"); ok && len(val.([]interface{})) > 0 {
+		noResponseData := val.([]interface{})[0].(map[string]interface{})
+		profile.NoResponse = getRDataInfo(noResponseData)
+	}
+
+	if val, ok := rd.GetOk("rdata_info"); ok {
+		rdataInfoDataList := val.(*schema.Set).List()
+		profile.RDataInfo = getRDataInfoList(rdataInfoDataList)
+	}
+
+	return rrSetData
+}
+
+func getRDataInfoList(rdataInfoDataList []interface{}) []*dirpool.RDataInfo {
+	rdataInfoList := make([]*dirpool.RDataInfo, len(rdataInfoDataList))
+
+	for i, d := range rdataInfoDataList {
+		rdataInfoData := d.(map[string]interface{})
+		rdataInfoList[i] = getRDataInfo(rdataInfoData)
+	}
+
+	return rdataInfoList
+}
+
+func getRDataInfo(rdataInfoData map[string]interface{}) *dirpool.RDataInfo {
+	rdataInfo := &dirpool.RDataInfo{}
+	geoInfo := &dirpool.GEOInfo{}
+	ipInfo := &dirpool.IPInfo{}
+
+	rdataInfo.GeoInfo = geoInfo
+	rdataInfo.IPInfo = ipInfo
+
+	if val, ok := rdataInfoData["type"]; ok {
+		rdataInfo.Type = val.(string)
+	}
+
+	if val, ok := rdataInfoData["ttl"]; ok {
+		rdataInfo.TTL = val.(int)
+	}
+
+	if val, ok := rdataInfoData["all_non_configured"]; ok {
+		rdataInfo.AllNonConfigured = val.(bool)
+	}
+
+	if val, ok := rdataInfoData["geo_group_name"]; ok {
+		geoInfo.Name = val.(string)
+	}
+
+	if val, ok := rdataInfoData["geo_force_overlap"]; ok {
```

Comment:
`geo_force_overlap` didn't make it into the documentation - is this intentional?

Reply:
geo_force_overlap should be removed from schema